app/js/settings.ts

```diff
@@ -23,8 +24,12 @@ const Settings = {
     filterGroupedTabs: false,
 
     // An array of tabids which have been explicitly locked by the user.
+    // TODO: rename to `lockedTabIds`?
     lockedIds: defaultLockedIds,
 
+    // An array of window ids which have been explicitly locked by the user.
+    lockedWindowIds: defaultLockedWindowIds,
```

Comment:
You should add a listener for [`windows.onRemoved`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/windows/onRemoved) and remove that window from the array. On startup of Tab Wrangler, you should potentially also get a list of all window IDs and remove any IDs from `lockedWindowIds` that no longer exist.

Those two bits combined will ensure `lockedWindowIds` doesn't grow forever.

Without this, for example, you could lock a window and then close it, and do that hundreds/thousands of times over the install period of Tab Wrangler and this array will continue to grow and take up space in the extension's limited storage.